node/cn/peer_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2019 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package cn
+
+import (
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/networks/p2p"
+	"github.com/stretchr/testify/assert"
+	"math/big"
+	"testing"
+)
+
+var version = 63
+
+func newBasePeer() (Peer, *p2p.MsgPipeRW, *p2p.MsgPipeRW) {
+	pipe1, pipe2 := p2p.MsgPipe()
+
+	return newPeer(version, p2pPeers[0], pipe1), pipe1, pipe2
+}
+
+func TestBasePeer_AddToKnownBlocks(t *testing.T) {
+	basePeer, _, _ := newBasePeer()
+	assert.False(t, basePeer.KnowsBlock(hash1))
+	basePeer.AddToKnownBlocks(hash1)
+	assert.True(t, basePeer.KnowsBlock(hash1))
+}
+
+func TestBasePeer_AddToKnownTxs(t *testing.T) {
+	basePeer, _, _ := newBasePeer()
+	assert.False(t, basePeer.KnowsTx(hash1))
+	basePeer.AddToKnownTxs(hash1)
+	assert.True(t, basePeer.KnowsTx(hash1))
+}
+
+func TestBasePeer_Send(t *testing.T) {
+	basePeer, _, oppositePipe := newBasePeer()
+	data := "a message data"
+	expectedMsg := generateMsg(t, NewBlockHashesMsg, data)
+	go func(t *testing.T) {
+		if err := basePeer.Send(NewBlockHashesMsg, data); err != nil {
+			t.Fatal(err)
+		}
+	}(t)
+	receivedMsg, err := oppositePipe.ReadMsg()
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(t, expectedMsg.Code, receivedMsg.Code)
+	assert.Equal(t, expectedMsg.Size, receivedMsg.Size)
```

Comment:
How about compare payload also?

https://github.com/klaytn/klaytn/blob/fc83163bf5f18c7f536932435c27d4f979dd0640/networks/p2p/message.go#L226-L252

Reply:
Good point! I will add more lines for that.